impl/maven-impl/src/main/java/org/apache/maven/impl/DefaultModelXmlFactory.java

```diff
@@ -48,8 +48,37 @@
 @Singleton
 public class DefaultModelXmlFactory implements ModelXmlFactory {
     @Override
+    @Nonnull
     public Model read(@Nonnull XmlReaderRequest request) throws XmlReaderException {
         nonNull(request, "request");
+        Model model = doRead(request);
+        if (isModelVersionGreaterThan400(model)
+                && !model.getNamespaceUri().startsWith("http://maven.apache.org/POM/")) {
```

Comment:
Will getNamespaceUri return an empty string (i,.e. not null) when there is no namespace?

Reply:
Yes, there are unit tests to support that.